run-tests.php

```diff
@@ -2780,9 +2780,14 @@ function run_test(string $php, $file, array $env): string
 
         // write .sh
         if (strpos($log_format, 'S') !== false) {
+            $env_lines = [];
+            foreach ($env as $env_var => $env_val) {
+                $env_lines[] = "export $env_var=" . escapeshellarg($env_val);
+            }
+            $exported_environment = $env_lines ? "\n" . implode("\n", $env_lines) . "\n" : "";
```

Comment:
The first `"\n"` here looks like too much?

Reply:
No it isn't, it will ensure a single blank line between shebang and exports.